signalrkore/src/commonMain/kotlin/eu/lepicekmichal/signalrkore/HubConnection.kt

```diff
@@ -400,12 +406,51 @@ class HubConnection private constructor(
         }
     }
 
+    private suspend fun processReceivedInvocation(message: HubMessage.Invocation) {
+        if (message is HubMessage.Invocation.Blocking && message.invocationId != null && subscribersWithResult[message.target] != true) {
+            logger.log(Logger.Level.ERROR, "Failed to find a value returning handler for ${message.target} method. Sending error to server.")
+
+            val resultMessage = HubMessage.Completion.Error(invocationId = message.invocationId, error = "Client did not provide a result.")
+            sendHubMessage(resultMessage)
+        }
+
+        receivedInvocations.emit(message)
+    }
+
     @OptIn(InternalSerializationApi::class)
     override fun <T : Any> T.toJson(kClass: KClass<T>): JsonElement = json.encodeToJsonElement(kClass.serializer(), this)
 
     @OptIn(InternalSerializationApi::class)
     override fun <T : Any> JsonElement.fromJson(kClass: KClass<T>): T = json.decodeFromJsonElement(kClass.serializer(), this)
 
+    override suspend fun <TResult> handleInvocation(
+        message: HubMessage.Invocation,
+        resultType: KClass<TResult>,
+        callback: suspend () -> TResult
+    ) where TResult : Any {
+        val invocationId = if (message is HubMessage.Invocation.Blocking) message.invocationId else null
+        var resultMessage: HubMessage.Completion? = null
+
+        try {
+            val result = callback()
+            if (invocationId != null) {
+                resultMessage = HubMessage.Completion.Resulted(invocationId = invocationId, result = result.toJson(resultType))
+            }
+        } catch (ex: Exception) {
+            logger.log(Logger.Level.ERROR, "Invoking client side method '${message.target}' failed: $ex")
```

Comment:
In Java, the exception is passed as a separate argument, but this is not possible with the current logger interface. Having  the ability to pass the exception as a separate argument would be beneficial when using the Android `Log.e()` method